File: Data Structures & Algorithms/is-anagram/submission-1.rs
use std::collections::hash_map::Entry;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false
        }
        let mut map = HashMap::new();
        for c in s.chars() {
            *map.entry(c).or_insert(0) += 1;
        }

        for c in t.chars() {
            match map.get_mut(&c) {
                Some(count) if *count > 0 => *count -= 1,
                _ => return false
            }
        }

        true
    }
}

File: Data Structures & Algorithms/hand-of-straights/submission-1.java
class Solution {
    public boolean isNStraightHand(int[] hand, int groupSize) {
        int[] count = new int[1001];
        for (int h : hand) {
            count[h]++;
        }
        for (int i = 0; i < count.length; ) {
            if (count[i] == 0) {
                i++;
                continue;
            }
            for (int j = 0; j < groupSize; j++) {
                if (count[i+j] == 0) return false;
                count[i+j]--;
            }
        }
        return true;
   }
}
